src/main/groovy/org/pih/warehouse/inventory/CycleCountRequestCommand.groovy

```diff
@@ -0,0 +1,25 @@
+package org.pih.warehouse.inventory
+
+import grails.validation.Validateable
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.product.Product
+
+class CycleCountRequestCommand implements Validateable {
+
+    Location facility
+
+    Product product
+
+    Boolean blindCount
+
+    static constraints = {
+        product(nullable: true, validator: { Product product, CycleCountRequestCommand obj ->
+            // TODO: Most probably, this will have to be modified not to include completed cycle counts
+            CycleCountRequest cycleCountRequest = CycleCountRequest.findByProductAndFacility(product, obj.facility)
```

Comment:
~Again the same question, did we want to delete the requests after completing, or should we look here for only `PENDING` Requests to find duplicates~ NVM, I just read the TODO, but the other constraint in the CCR domain itself need either improvement or TODO as well.

Reply:
But I think you could already use the `findByProductAndFacilityAndStatusNotInList(product, obj.facility, [COMPLETED, CANCELED])`